### PR TITLE
Rough notes as basis for initial dev guidelines 

### PR DESCRIPTION
**Pull request by ibgreen**
Captures notes in other tickets, intended to be updated/cleaned up based on further discussions.

### CODE REVIEW

File: dev-guidelines/common/DEPRECATION-GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+* In cases where we can't or won't do gentle deprecation, we should document the breaking changes in an upgrade guide.
+
+* I'm in favor of following what Facebook does with React in general - console.warn for API deprecations and clearly stating in the CHANGELOG and then deprecate 2-3 versions later.
+
+* The question also is when to eventually do the deprecation. Do we do it in a minor patch, or wait for a major version bump?
+
+ * iven we already have a few majors, everything that is potentially breaking should be done in a major
+ @ibgreen
+
+* Isn't the key consideration that we should provide some reasonable time for users to update? If a feature has been deprecated for 6+ months, and we are cutting a new major release, it would seem acceptable to remove it.
```
Comment (shaojingli) [inline dev-guidelines/common/DEPRECATION-GUIDELINES.md]: This paragraph sounds like a note... we should rephrase the first sentence (the question) as a statement.

File: dev-guidelines/common/DEPRECATION-GUIDELINES.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+* If it was just deprecated a week earlier in a minor release, and we are cutting a new major release, then we should probably keep it around as deprecated for another major release.
+
+* Even if it has only been deprecated for a week and we are about to draft a new major, we should take the opportunity to remove it completely as soon as possible. If users want to update to the latest major and stay 
+* gently deprecate with console.warn when possible (and not too burdensome)
+
+
+* note API deprecation and removal in CHANGELOG
```
Comment (shaojingli) [inline dev-guidelines/common/DEPRECATION-GUIDELINES.md]: This paragraph is not complete

File: dev-guidelines/common/TEST-GUIDELINES.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+
+## Node tests
+
+
+Idea behind `test/node.js` and `test/node-dist.js`
+
+These are supposed to work identically. The only difference being that one runs the test directly on the ES6 source and one runs the tests on the transpiled ES5 version of the library, just to make doubly sure the transpilation doesn't cause any issues. We should absolutely not import jsdom in just one of these two tests, they should both have the same setup, either neither or both should import it. Second, about jsdom usage in node tests:
+
+The big problem is that the introduction of jsdom hides an important class of problems - i.e. we will no longer catch issues that happen under node in apps that don't include jsdom - which are very common (stray references to window and document etc). Thus, importing jsdom defeats an important part of the purpose of the node tests.
```
Comment (shaojingli) [inline dev-guidelines/common/TEST-GUIDELINES.md]: JSDom has been removed. So we should remove this paragraph or rephrase it.

**Comment by ibgreen** [comment]
> Let's merge this in and I'll update it according to my comments later.

Yep that was the idea 👍

**Pull request merged**